packages/server/src/api/controllers/application.ts

```diff
@@ -280,13 +288,41 @@ exports.create = async ctx => {
   }
 
   await appCache.invalidateAppMetadata(appId, newApplication)
-  ctx.status = 200
+  return newApplication
+}
+
+const appPostCreate = async (ctx: any, appId: string) => {
+  // app import & template creation
+  if (ctx.request.body.useTemplate === "true") {
+    const rows = await getUniqueRows([appId])
+    const rowCount = rows ? rows.length : 0
+    if (rowCount) {
+      try {
+        await quotas.addRows(rowCount)
+      } catch (err: any) {
+        if (err.code && err.code === errors.codes.USAGE_LIMIT_EXCEEDED) {
```

Comment:
NAB: Would be good to send something to posthog here.

`budibase:row_limit_exceeded`

Reply:
good idea! this could go in the usage checker itself so we get it for all of them. could then use a namespace like
`budibase:quota_exceeded:row`
`budibase:quota_exceeded:apps`
etc etc